src/Analyzer.PowerShellRuleEngine/PowerShellRuleEngine.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.Templates.Analyzer.Types;
+
+namespace Microsoft.Azure.Templates.Analyzer.RuleEngines.PowerShellEngine
+{
+    /// <summary>
+    /// Executes template analysis encoded in PowerShell
+    /// </summary>
+    public class PowerShellRuleEngine
+    {
+        /// <summary>
+        /// Execution environment for PowerShell
+        /// </summary>
+        private readonly System.Management.Automation.PowerShell PowerShell;
+
+        /// <summary>
+        /// Regex that matches a string like: " on line: aNumber"
+        /// </summary>
+        private readonly Regex LineNumberRegex = new(@"\son\sline:\s\d+", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+        /// <summary>
+        /// Creates a new instance of a PowerShellRuleEngine
+        /// </summary>
+        public PowerShellRuleEngine()
+        {
+            this.PowerShell = System.Management.Automation.PowerShell.Create();
+
+            PowerShell.Streams.Error.DataAdded += (sender, e) => HandleDataAddedInStreams(PowerShell.Streams.Error[e.Index], ConsoleColor.Red);
+            PowerShell.Streams.Warning.DataAdded += (sender, e) => HandleDataAddedInStreams(PowerShell.Streams.Warning[e.Index], ConsoleColor.Yellow);
+            PowerShell.Streams.Information.DataAdded += (sender, e) => HandleDataAddedInStreams(PowerShell.Streams.Information[e.Index]);
+            PowerShell.Streams.Verbose.DataAdded += (sender, e) => HandleDataAddedInStreams(PowerShell.Streams.Verbose[e.Index]);
+            PowerShell.Streams.Debug.DataAdded += (sender, e) => HandleDataAddedInStreams(PowerShell.Streams.Debug[e.Index]);
+
+            PowerShell.Commands.AddCommand("Set-ExecutionPolicy")
+                .AddParameter("Scope", "Process") // Affects only the current PowerShell session
+                .AddParameter("ExecutionPolicy", "Unrestricted");
+            PowerShell.AddStatement();
+
+            PowerShell.Commands.AddCommand("Import-Module")
+                .AddParameter("Name", @".\TTK\arm-ttk.psd1"); // arm-ttk is added to the needed project's bins directories in build time 
```

Comment:
This should provide the full absolute path (not relative path) to the TTK module.  Otherwise, if the working directory is different than BPA's location when BPA is executed, ".\TTK" won't be found.

You can get the directory BPA is in by
- Getting the location of the PS engine dll: `Assembly.GetExecutingAssembly().Location` (in the System.Reflection namespace)
- Using path helper functions in System.IO.Path with the path of the dll.

Reply:
Good catch, thank you